subhub/hub/stripe/customer.py

```diff
@@ -198,29 +203,56 @@ def run(self):
                     customer_id=self.payload.data.object.customer,
-                    subscriptionId=self.payload.data.object.id,  # required by FxA
                     subscription_id=self.payload.data.object.id,
                     plan_amount=self.payload.data.object.plan.amount,
                     canceled_at=self.payload.data.object.canceled_at,
                     cancel_at=self.payload.data.object.cancel_at,
                     cancel_at_period_end=self.payload.data.object.cancel_at_period_end,
                     nickname=self.payload.data.object.plan.nickname,
-                    messageCreatedAt=int(time.time()),  # required by FxA
                     invoice_id=self.payload.data.object.latest_invoice,
-                    eventId=self.payload.id,  # required by FxA
+                    current_period_start=self.payload.data.object.current_period_start,
+                    current_period_end=self.payload.data.object.current_period_end,
                 )
                 logger.info("customer subscription cancel at period end", data=data)
-                routes = [StaticRoutes.FIREFOX_ROUTE, StaticRoutes.SALESFORCE_ROUTE]
+                routes = [StaticRoutes.SALESFORCE_ROUTE]
                 self.send_to_routes(routes, json.dumps(data))
             elif (
                 not self.payload.data.object.cancel_at_period_end
                 and self.payload.data.object.status == "active"
-                and not previous_attributes.get("cancel_at_period_end")
+                # and not previous_attributes.get("cancel_at_period_end")
             ):
-                data = self.create_data(
+                invoice_number = None
+                invoice_id = None
+                charge = None
+                last4 = None
+                brand = None
+                try:
+                    customer_id = self.payload.data.object.customer
+                    updated_customer = stripe.Customer.retrieve(id=customer_id)
+                    user_id = updated_customer.metadata.get("userid")
+                    invoice_id = self.payload.data.object.latest_invoice
+                    latest_invoice = self.get_latest_invoice(invoice_id)
+                    logger.info("latest invoice", latest_invoice=latest_invoice)
+                    invoice_number = latest_invoice.number
+                    charge = latest_invoice.charge
+                    logger.info("charge", charge=charge)
+                    latest_charge = self.get_latest_charge(charge)
+                    logger.info("latest charge", latest_charge=latest_charge)
+                    last4 = latest_charge.payment_method_details.card.last4
+                    brand = latest_charge.payment_method_details.card.brand
+                except InvalidRequestError as e:
+                    logger.error("Unable to gather data", error=e)
+                    raise InvalidRequestError(
+                        message="Unable to gather data", param=str(e)
```

Comment:
actually why are we not just `raise`ing the exception after logging it?  `raise e`